internal/services/pam/pam.go

```diff
@@ -96,10 +96,14 @@ func (s Service) GetPreviousBroker(ctx context.Context, req *authd.GPBRequest) (
 		return &authd.GPBResponse{}, nil
 	}
 
-	// Updates manager memory to stop needing to query the database for the broker.
-	if err = s.brokerManager.SetDefaultBrokerForUser(brokerID, req.GetUsername()); err != nil {
-		log.Warningf(ctx, "Last broker used by %q is not available, letting the user selecting one: %v", req.GetUsername(), err)
-		return &authd.GPBResponse{}, nil
+	// Cache the broker which should be used for the user, so that we don't have to query the database again next time -
+	// except if the broker is the local broker, because then the decision to use the local broker should be made each
+	// time the user tries to log in, based on whether the user is provided by any other NSS service.
+	if brokerID != brokers.LocalBrokerName {
+		if err = s.brokerManager.SetDefaultBrokerForUser(brokerID, req.GetUsername()); err != nil {
+			log.Warningf(ctx, "Last broker used by %q is not available, letting the user selecting one: %v", req.GetUsername(), err)
+			return &authd.GPBResponse{}, nil
+		}
```

Comment:
What about just early returning:

```go
if brokerID == brokers.LocalBrokerName  {
        // Do not save the previous broker in this case because....
	return &authd.GPBResponse{
		PreviousBroker: brokerID,
	}, nil
}

Reply:
I think the `if brokerID != brokers.LocalBrokerName {` approach is a bit more future proof because if we add fields to the response or do anything else before returning, with your approach we might miss adapting the early return path as well. I don't care much though, so if you prefer the early return I'll do it.

Reply:
Done in a fixup commit